winsdk/cache.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"time"
+
+	"github.com/0chain/gosdk/zboxcore/sdk"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+type cachedAllocation struct {
+	Expiration time.Time
+	Allocation *sdk.Allocation
+}
+
+var (
+	cachedAllocations, _ = lru.New[string, *cachedAllocation](100)
+)
+
+func getAllocation(allocationId string) (*sdk.Allocation, error) {
+
+	var it *cachedAllocation
+	var ok bool
+
+	it, ok = cachedAllocations.Get(allocationId)
+
+	if ok {
+		if ok && it.Expiration.After(time.Now()) {
+			return it.Allocation, nil
+		}
+	}
+
+	a, err := sdk.GetAllocation(allocationId)
+	if err != nil {
+		return nil, err
+	}
+
+	it = &cachedAllocation{
+		Allocation: a,
+		Expiration: time.Now().Add(5 * time.Minute),
```

Comment:
it confuses a little bit, since allocation has it's own expiration, maybe we can have some different wording for it

Reply:
fixed